theano/sandbox/cuda/opt.py

```diff
@@ -1589,19 +1589,20 @@ def local_conv_gemm(node):
         img, kern = node.inputs
         border_mode = node.op.border_mode
         subsample = node.op.subsample
-        if (border_mode == 'valid') or (subsample != (1, 1)):
+        filter_dilation = node.op.filter_dilation
+        if (border_mode == 'valid') or (subsample != (1, 1)) or filter_dilation != (1, 1):
             # need to flip the kernel for valid convolution
             kern = kern[:, :, ::-1, ::-1]
             # By default use GpuCorrMM
-            rval = GpuCorrMM(border_mode, subsample)(
+            rval = GpuCorrMM(border_mode, subsample, filter_dilation)(
                 gpu_contiguous(img), gpu_contiguous(kern))
 
             # call GpuCorrMM_gradWeights if good
             # (the latter is faster if batchsize * kernelHeight * kernelWidth
             # is larger than inputChannels * outputHeight * outputWidth.
             # GpuConv does not always store information on the batchsize and
             # channels, though, so we only use what information we have.)
-            if ((subsample == (1, 1)) and
+            if ((subsample == (1, 1)) and filter_dilation == 1 and
```

Comment:
Shouldn't this be `(1, 1)`? Is that normalized to become a 2-tuple in the constructor?